lib/pi_piper/bcm2835.rb

```diff
@@ -36,7 +36,8 @@ def self.pin_input(pin)
     end
 
     def self.pin_set(pin, value)
-      File.open("/sys/class/gpio/gpio#{pin}/value", 'w') {|f| f.write("#{value}") }
+      raise PiPiper::PinError, "Pin #{pin} not exported" unless self.exported?(pin)
```

Comment:
maybe, but it works as is. nevermind we can have beautiful code in the rewrite branch.